### ISSUE TITLE
Form validation on URL entries: error message/alert banner

### ISSUE THREAD

**Issue by eporter23** [emory-libraries/dlp-curate#1308]
For form fields that enforce valid, fully qualified URLs as values, the following enhancements are recommended:

When a user attempts to save changes to a work which include an improperly formatted URL entry, the user doesn't receive a warning until they submit a save request. This is different behavior than occurs with Date fields, which show validation errors as a user types in the form field.

The following error message occurs near the form field but only after the user has clicked Save Changes. In some cases, this field may also be in the initially hidden field grouping and is no longer visible.

![Screen Shot 2020-06-30 at 3.16.25 PM.png](https://images.zenhubusercontent.com/5b6de3fda32a4f476097b0be/57ac79a3-9230-45e4-83e4-f2e6bc6bf4e3)

Additionally, when a URI field fails validation, a red banner appears at the top of the form, but no text is displayed in the banner.

![Screen Shot 2020-06-30 at 3.13.39 PM.png](https://images.zenhubusercontent.com/5b6de3fda32a4f476097b0be/22c3092d-7cb5-421e-be6c-1f0afdb1fc02)

Desired behavior:
- [ ] Display a validation warning with red text below the form field (like Date entries do) indicating "This field requires a complete and valid URL"
- [ ] Populate OR remove display of empty red alert bar a the top of the form. If populating the alert, the text should display the field label for where the validation failed plus a generic message: "One or more field entries require a complete and valid URL. Please check your submissions." 

Optional/later:
- [ ] Disable the _Describe your work_ green check in the Save Work sidebar and prohibit Save until the error is corrected and the user has exited the field

**Comment by eporter23** [comment]
Discussion: it will be easier to populate the banner than do JS validation.

**Comment by eporter23** [comment]
@devanshu-m this worked fine in test, but in prod if I submit an invalid entry in a URL field I get a generic application error instead: 

![Screen Shot 2020-07-31 at 5.38.01 PM.png](https://images.zenhubusercontent.com/5b6de3fda32a4f476097b0be/c870bab4-b237-4239-ad0f-bbc47fef8f9b)

**Comment by eporter23** [comment]
Update: this occurs when creating a brand new work that contains invalid URLs. It works as expected when editing existing works.

**Comment by eporter23** [comment]
@devanshu-m I just re-tested this in test, but if I create a new work, add an invalid URL entry, I get the prompt to correct it, and I am taken back to the Descriptions tab.

At the point a couple of things happen:
1) the attached fileset is lost (must be recreated)
2) when I try to correct the URL entry with a valid URL, the error does not dismiss and I can't save.

![Screen Shot 2020-08-06 at 4.24.55 PM.png](https://images.zenhubusercontent.com/5b6de3fda32a4f476097b0be/c3373a8c-bdc2-4af0-b028-4166aa88eb85)

**Comment by devanshu-m** [comment]
@eporter23 I will need to take a look at these errors and might take a while for me to find fixes. Do we want to create a new ticket or re-estimate this one? This might also mean changes to where the preservation_event for work creation/policy assignment, etc. happen.

**Comment by devanshu-m** [comment]
Research notes (since I am out tomorrow):
There is some weirdness going on with work creation - when we try creating a work through the work form and enter a bad url, the validation fails, and we are taken back to the work creation page. On correcting the bad url and re-submitting, there are two things that happen:
1. A `permissions_attributes` field is added to the params, which is normal hyrax behavior, however, in hyrax this field is added when editing a work and not creating a work. Because this field is not handled by hyrax during work creation, we get the following error:
![image.png](https://images.zenhubusercontent.com/5c12bc2dd3542818d52aa3a0/1d0aad08-e6b1-463c-8196-88d8be3806de)
The error we are getting seems to be a known issue: https://samvera.slack.com/archives/C0F9JQJDQ/p1528485794000143?thread_ts=1528481893.000255&cid=C0F9JQJDQ

The solution I came up with was clearing out the `permissions_attributes` from the params during work creation in the `CurateGenericWorkActor`. I tested this manually and this seems to work. However, I feel nervous with this change since we are altering how works get created.
2. All uploaded files are lost.

Now the reason all of this might be happening in the first place is because we are doing validations in the model which Hyrax is not used to. What we can do here, is try and change our validation strategy. This will require more discussion. I will reach out Emily on Monday.

I have done a little bit of research in samvera and samvera slack and do not seem to find any references of validations happening in models.

**Comment by eporter23** [comment]
@devanshu-m I agree that we should hold off on any major changes to work creation and discuss first. Also, since these errors only occur in very specific circumstances while _creating_ a new work, I think we can also consider resolving this through training/support. The URL validation you previously established is working fine when editing works which I think is probably more important.

**Comment by eporter23** [comment]
I confirmed this again with @devanshu-m - the validation work already implemented is working very well except for in that unusual circumstance about when creating a brand new work. Rather than doing significant alterations to any existing create work processes, it'd be easier and safer to just document this as a known issue and train Curate users in this. As we discussed, it is an edge case that would be very infrequent. Also, if user encounters this they can just start over. We may also consider making URL validation Javascript based instead of application based. If we do this, we would look at reverting the code implemented here.
